test/run.py

```diff
@@ -353,9 +353,11 @@ def collect_dir_tests(base_dir, test_files, check_thirdparty=False):
             path = os.path.join(base_dir, f_name)
 
             if is_py3:
-                source = open(path, encoding='utf-8').read()
+                f = open(path, encoding='utf-8')
```

Comment:
If we're already cleaning up messes, please use `with open(path, encoding='utf-8') as f`. My bad for not doing that.

Reply:
Done.